src/Disks/ObjectStorages/MetadataStorageFromPlainObjectStorage.h

```diff
@@ -20,7 +21,7 @@ namespace DB
 /// It is used to allow BACKUP/RESTORE to ObjectStorage (S3/...) with the same
 /// structure as on disk MergeTree, and does not requires metadata from local
 /// disk to restore.
-class MetadataStorageFromPlainObjectStorage final : public IMetadataStorage
+class MetadataStorageFromPlainObjectStorage final : public ReadOnlyMetadataStorage
```

Comment:
Why is it inherited from ReadOnlyMetadataStorage? I mean `MetadataStorageFromPlainObjectStorage` doesn't look like it should be readonly. Probably a comment is needed.

Reply:
Yes, it is not completely readonly, but it does not have any extended metadata support (chown/mtime/stat/...) and `ReadOnlyMetadataStorage` throws `NOT_IMPLEMENTED` exception for this API.

I thought about the name for `ReadOnlyMetadataStorage`, and everything else looks more complex and less cleaner.

Maybe you have any other ideas?

>Probably a comment is needed

Ok (at first I had a comment for ReadOnlyMetadataTransaction/ReadOnlyMetadataStorage but eventually decided that it will only confuse reader)